fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -496,6 +575,45 @@ public void setCurrentLaoName(String laoName) {
         return mIsOrganizer;
     }
 
+    public LiveData<Boolean> isWitness() {
+        try {
+            KeysetHandle publicKeysetHandle =
+                    mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+            boolean isWitness =
+                    getCurrentLaoValue().getWitnesses().contains(Keys.getEncodedKey(publicKeysetHandle));
+            Log.d(TAG, "isWitness: " + isWitness);
+            mIsWitness.setValue(isWitness);
```

Comment:
The thing is that the isOrganizer will never change once the lao is already created , but for the witness suppose we scan a new witness after the lao has already been subscribed to then isWitness() will return false every time . Maybe a more compliant solution would be to update mIsWitness when we get the result of updateLaoWitnesses() ?

Reply:
Yes that sounds like a good solution

Reply:
Actually it wouldn't work because even in updateLaoWitnesses the new witness list  is only being updated in handleStateLao of LaoRepository after the broadcast has been received . And because we can't have access to laoDetailViewModel.setWitness() in the LaoRepository I don't really see where we could set the witnesses elsewhere than here .